### DISCUSSION TITLE
Cannot call default constructor from __init__ if Base class not created with declarative_base()

### DISCUSSION THREAD

**Discussion by jdavcs** [sqlalchemy/sqlalchemy#7015] [Usage Questions] [answered]
### Describe the bug

I have a class mapped declaratively with the base class constructed explicitly (not using `declarative_base()`). I override the `__init__` constructor (to add additional initialization while taking advantage of the default constructor, so that I don't have to specify keyword parameters for all the fields). This does not work, because `super().__init__(**kwd)` does not call SA's default implementation. However, this works when I create the base class using `declarative_base()`. 

Both base classes have the same metaclass (obviously); however only the one created with declarative_base() has `sqlalchemy.orm.decl_api.Base` as its base class, which, I assume, explains why the call to `super().__init__()` works only for the latter.

I suspect this is not the intended behavior, since the documentation states that the two base classes are equivalent [ref](https://docs.sqlalchemy.org/en/14/orm/mapping_styles.html#creating-an-explicit-base-non-dynamically-for-use-with-mypy-similar). However, if it is expected, I'll work around it.

### To Reproduce

```python
from sqlalchemy import Column, Integer 
from sqlalchemy.orm import declarative_base, registry
from sqlalchemy.orm.decl_api import DeclarativeMeta

mapper_registry = registry()

class Base1(metaclass=DeclarativeMeta):
    __abstract__ = True 
    registry = mapper_registry
    metadata = mapper_registry.metadata
 
class A(Base1):
    __tablename__ = 'foo'
    id = Column(Integer, primary_key=True)
    a = Column(Integer)
    
    def __init__(self, **kwd):
        super().__init__(**kwd)
        # additional initialization    
     
Base2 = declarative_base()

class B(Base2):
    __tablename__ = 'bar'
    id = Column(Integer, primary_key=True)
    a = Column(Integer)
    
    def __init__(self, **kwd):
        super().__init__(**kwd)
        # additional initialization  

A(a=1)  # ERROR
B(a=1)  # OK
```

### Error

```
Traceback (most recent call last):
  File "/home/sergey/0dev/drafts/python/sqlalchemy/misc/demo.py", line 35, in <module>
    A(a=1)
  File "<string>", line 4, in __init__
  File "/home/sergey/0dev/drafts/python/sqlalchemy/misc/.venv/lib64/python3.9/site-packages/sqlalchemy/orm/state.py", line 480, in _initialize_instance
    manager.dispatch.init_failure(self, args, kwargs)
  File "/home/sergey/0dev/drafts/python/sqlalchemy/misc/.venv/lib64/python3.9/site-packages/sqlalchemy/util/langhelpers.py", line 70, in __exit__
    compat.raise_(
  File "/home/sergey/0dev/drafts/python/sqlalchemy/misc/.venv/lib64/python3.9/site-packages/sqlalchemy/util/compat.py", line 207, in raise_
    raise exception
  File "/home/sergey/0dev/drafts/python/sqlalchemy/misc/.venv/lib64/python3.9/site-packages/sqlalchemy/orm/state.py", line 477, in _initialize_instance
    return manager.original_init(*mixed[1:], **kwargs)
  File "/home/sergey/0dev/drafts/python/sqlalchemy/misc/demo.py", line 20, in __init__
    super().__init__(**kwd)
TypeError: object.__init__() takes exactly one argument (the instance to initialize)
```

### Versions

- OS: Fedora 34
- Python: 3.9.6
- SQLAlchemy: 1.4.22
- Database: -
- DBAPI: -

**Comment by CaselIT** [comment, marked as answer]
Hi,

I believe this is the expect behaviour. The documentation probably meant that the two versions are equivalent regarding registration of orm models.

If you remove the init from A does it work correctly, or a proper also init fails to generate?

**Reply by jdavcs** [reply]
Without the `__init__` it works correctly, of course. It's only the call to `super().__init__()` that doesn't work as I expected in that particular case. If that's expected behavior, that's fine. Thank you for the fast reply!

**Reply by zzzeek** [reply]
hi -

this is a documentation bug.  create the base like this:

```py
class Base(metaclass=DeclarativeMeta):
    __abstract__ = True

    # these are supplied by the sqlalchemy2-stubs, so may be omitted
    # when they are installed
    registry = mapper_registry
    metadata = mapper_registry.metadata

    __init__ = mapper_registry.constructor

```


will update docs

**Reply by zzzeek** [reply]
fixed in 7d4367602a05249232d49dbfd47

**Reply by jdavcs** [reply]
@zzzeek Many thanks! That takes care of it!
